nextjs/src/app/reports/[id]/(components)/MapLayers/PoliticalChoropleths.tsx

```diff
@@ -38,11 +38,22 @@ const PoliticalChoropleths: React.FC<PoliticalChoroplethsProps> = ({
       ? 'visible'
       : 'none'
 
+  const visualisationType =
+    report.displayOptions?.dataVisualisation?.visualisationType
+
   const { data: dataByBoundary } = useDataByBoundary({ report, boundaryType })
   const map = useLoadedMap()
   const [selectedBoundary, setSelectedBoundary] = useAtom(selectedBoundaryAtom)
   useClickOnBoundaryEvents(visibility === 'visible' ? tileset : null)
 
+  const [fillVisibility, setFillVisibility] = useState<'choropleth' | 'none'>(
+    visualisationType === 'none' ? 'none' : 'choropleth'
+  )
+
+  useEffect(() => {
+    setFillVisibility(visualisationType === 'none' ? 'none' : 'choropleth')
+  }, [visualisationType])
```

Comment:
question: do you need the extra state here? When you change the `visualisationType` in the dropdown, does it not update `report.displayOptions?.dataVisualisation?.visualisationType` ?